src/query/graphite/graphite/retention_policies.go

```diff
@@ -0,0 +1,106 @@
+package graphite
+
+import (
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/m3db/m3/src/query/graphite/ts"
+)
+
+const (
+	metricsTypeNodeIdx = 2
+	timerCountSuffix   = ".count"
+)
+
+// A RetentionPeriod emulates a Graphite retention period on top of M3,
+// consolidating raw datapoints into a defined time window using a specific
+// consolidation function.
+type RetentionPeriod struct {
+	pattern  *regexp.Regexp
+	policies []*RetentionPolicy
+}
+
+// A RetentionPolicy describes how Graphite aggregates a given metric within a
+// given archive
+type RetentionPolicy struct {
+	TTL           time.Duration
+	UnitPerStep   time.Duration
+	Consolidation ts.ConsolidationApproach
+	RoundToUnit   bool
+}
+
+var (
+	// NB(mmihic): These need to be ordered
+	graphiteRetentionPeriods = []*RetentionPeriod{
```

Comment:
Don't think we need any of these retention period definitions.

Let me see where this tricks up from, I'm pretty certain we can remove them.